csaf_2.0/prose/csaf-v2-editor-draft.md

```diff
@@ -2278,6 +2283,69 @@ The Weakness name (`name`) has value type `string` with 1 or more characters and
 
 Discovery date (`discovery_date`) of value type `string` with format `date-time` holds the date and time the vulnerability was originally discovered.
 
+#### 3.2.3.XYZ Vulnerabilities Property - Flags
+
+List of flags (`flags`) of value type `array` with 1 or more items of type `object` contains a list of machine readable flags.
+
+```
+    "flags": {
+      // ...
+      "items": {
+        // ...
+      }
+    },
+```
+
+Every Flag item of value type `object` with the mandatory property Label (`label`) contains product specific information in regard to this vulnerability as a single machine readable flag.
+For example, this could be a machine readable justification code why a product is not affected.
+
+> These flags MAY be used to decide in an automated fashion which actions to take.
+
+In addition, any Flag items MAY provide the three optional properties Date (`date`), Group IDs (`group_ids`) and Product IDs (`product_ids`).
+
+```
+    "properties": {
+      "date": {
+        // ...
+      },
+      "group_ids": {
+        // ...
+      },
+      "label": {
+        // ...
+      },
+      "product_ids": {
+        // ...
+      }
+    }
+```
+
+Date of the flag (`date`) of value type `string` with format `date-time` contains the date when assessment was done or the flag was assigned.
+
+Group IDs (`group_ids`) are of value type Product Groups (`product_groups_t`).
+
+Label of the flag (`label`) of value type `string` as `enum` specifies the machine readable label. Valid `enum` values are:
+
+```
+    component_not_present
+    inline_mitigations_already_exist
+    vulnerable_code_cannot_be_controlled_by_adversary
+    vulnerable_code_not_in_execute_path
+    vulnerable_code_not_present
+```
+
+The given values reflect the VEX not affected justifications. See [VEX-Just] for more details. The values MUST be used as follows:
+
+* `component_not_present`: The software is not affected because the vulnerable component is not in the product.
+* `vulnerable_code_not_present`: The product is not affected because the code underlying the vulnerability is not present in the product.
+  > Unlike `component_not_present`, the component in question is present, but for whatever reason (e.g. compiler options) the specific code causing the vulnerability is not present in the component.
+* `vulnerable_code_cannot_be_controlled_by_adversary`: The vulnerable component is present, and the component contains the vulnerable code. However, vulnerable code is used in such a way that an attacker cannot mount any anticipated attack.
+* `vulnerable_code_not_in_execute_path`: The affected code is not reachable through the execution of the code, including non-anticipated states of the product.
+  > Many products contain additional components that are not used or executed by the application.
```

Comment:
Maybe replace `Many products contain additional components` with `Components`?

`Many` Reads a bit like sales lingo to me. Also, no need for product, might be a tool, package, module, … that has a name, and dead code within.

Reply:
Sorry - I'm currently stuck:
Your suggestion was:
```suggestion
  > Components that are not used or executed by the application.
```
Is that correct?

Reply:
BTW: I would that exchange `application` against `product` as everything is a product in CSAF :wink: